chainer/functions/util/forget.py

```diff
@@ -105,6 +108,10 @@ def forget(func, *xs):
         This conversion takes place to ensure that this function is included
         in the computational graph to enable backward computations.
 
+    .. note::
+
+        ``F.forget`` does not support double backpropagation.
```

Comment:
Can we raise an error when users try to use `forget` with double backprop, e.g. by checking `not chainer.config.enable_backprop`?

Reply:
I added code that raises an error when `chainer.config.enable_backprop == True`. I added a test as well.